### PR TITLE
client: remove isTruthy and isFalsy

### PR DESCRIPTION
**Pull request by gabrocheleau**
Extracts client package from #2233

### CODE REVIEW

**Review by jochem-brouwer — COMMENTED** [review]
One remark

File: packages/client/lib/net/protocol/protocol.ts [hunk 1]
```diff
@@ -74,12 +72,12 @@ export class Protocol {
...
       }, this.timeout)
       const handleStatus = (status: any) => {
-        if (isTruthy(timeout)) {
+        if (timeout !== null && timeout !== 0) {
           clearTimeout(timeout)
           resolve(this.decodeStatus(status))
         }
       }
-      if (isTruthy(sender.status)) {
+      if (sender.status !== undefined && sender.status !== null) {
```
[the remark this answers is not in the document]
Comment (gabrocheleau) [reply packages/client/lib/net/protocol/protocol.ts]: Not 100% sure, but added it for safety. A lot of those helpers are untyped so hard to follow what's going on, better to err on the side of safety!

**Review by jochem-brouwer — APPROVED** [review]
LGTM, all is OK if CI passes :+1:

**Review by acolytec3 — APPROVED** [review]
LGTM!

**Comment by acolytec3** [comment]
Okay, I've been over it again and looks good.  Any reason not to merge assuming the CI passes.
